x-pack/plugins/reporting/server/config/index.ts

```diff
@@ -32,42 +32,6 @@ export const config: PluginConfigDescriptor<ReportingConfigType> = {
     unused('capture.viewport', { level: 'warning' }), // deprecated as unused since 7.16
     (settings, fromPath, addDeprecation) => {
       const reporting = get(settings, fromPath);
-      if (reporting?.roles?.enabled !== false) {
-        addDeprecation({
-          configPath: `${fromPath}.roles.enabled`,
-          level: 'warning',
-          title: i18n.translate('xpack.reporting.deprecations.reportingRoles.title', {
-            defaultMessage: `The "{fromPath}.roles" setting is deprecated`,
-            values: { fromPath },
-          }),
-          // TODO: once scheduled reports is released, restate this to say that we have no access to scheduled reporting.
-          // https://github.com/elastic/kibana/issues/79905
-          message: i18n.translate('xpack.reporting.deprecations.reportingRoles.description', {
-            defaultMessage:
-              `The default mechanism for Reporting privileges will work differently in future versions,` +
-              ` which will affect the behavior of this cluster. Set "xpack.reporting.roles.enabled" to` +
-              ` "false" to adopt the future behavior before upgrading.`,
-          }),
-          correctiveActions: {
-            manualSteps: [
-              i18n.translate('xpack.reporting.deprecations.reportingRoles.manualStepOne', {
-                defaultMessage: `Set "xpack.reporting.roles.enabled" to "false" in kibana.yml.`,
-              }),
-              i18n.translate('xpack.reporting.deprecations.reportingRoles.manualStepTwo', {
-                defaultMessage: `Remove "xpack.reporting.roles.allow" in kibana.yml, if present.`,
-              }),
-              i18n.translate('xpack.reporting.deprecations.reportingRoles.manualStepThree', {
-                defaultMessage:
-                  `Go to Management > Security > Roles to create one or more roles that grant` +
-                  ` the Kibana application privilege for Reporting.`,
-              }),
-              i18n.translate('xpack.reporting.deprecations.reportingRoles.manualStepFour', {
-                defaultMessage: `Grant Reporting privileges to users by assigning one of the new roles.`,
-              }),
-            ],
-          },
-        });
-      }
```

Comment:
The `index.test.ts` file needs to be updated here as well:
* `logs a warning of roles.enabled for defaults`: this test should be changed to test that no warnings are logged by default
* `logs a warning if roles.enabled: true is set`: this should be changed to test that a warning is logged if `xpack.reporting.roles.allow` is set to a non-empty value (because [we still need a warning if that setting is set](https://github.com/elastic/kibana/pull/206564/files?show-viewed-files=true&file-filters%5B%5D=#r1917346318)) 
* `does not log a warning recommended settings are used`: the `roles: { enabled: false },` line can be removed from the config object for this test